src/GameOfLifeBoard.js

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import next from './gol';
+import Board from './Board';
+import Button from './Button';
+
+class GameOfLifeBoard extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      boardState : props.boardState,
+      initialBoardState : props.boardState
+    };
+    this.handleNextClick = this.handleNextClick.bind(this); // the "handleNextClick" must be bound, or arro function would need to be used
+    this.handleResetClick = this.handleResetClick.bind(this);
+  }
+
+  handleNextClick(){
+    this.setState(currentState => { // State Mutation. First argument is always the whole current state
+      return { // Function must return the whole New State
+        ...currentState, // "..." (spread) is deconstruction of the object adding the change
+        boardState : next(currentState.boardState) // adding new state
+      }
+    });
+  }
+
+  handleResetClick(){
+    this.setState(currentState => {
+      return {
+        ...currentState,
+        boardState: this.state.initialBoardState
+      }
+    })
+  }
+
+  render() {
+    return (<div>
+      {this.state.boardState && <Board boardState={this.state.boardState}/>}
+      <br />
+      <Button onClick={this.handleNextClick} title="Další generace!"/>
+      <Button onClick={this.handleResetClick} title="Resetovat"/>
+    </div>)
+  }
+}
+
+export default GameOfLifeBoard;
```

Comment:
What if `GameOfLife` component will have some logic and changes `boardState` sent in props?

Reply:
I'm not sure if I'm following your question, @vysinsky. What if the `boardState` structure is different? I could use `PropSpec` to make sure that the initial state is according expectations. Truth is that I have not applied much of "defensive" programming in the pull request.

Are you suggesting something else?

Reply:
I mean something else. Let's talk about this tomorrow in person :).